### PR TITLE
Add eBay compliance fix

### PR DESCRIPTION
**Pull request by craiga**
eBay compliance fix as discussed in #403.

Thanks to @thiras and @sambragg for working out what was going on.

This will hopefully make it easier for the next developer who needs to figure this out.

### CODE REVIEW

**Review by heyscam — APPROVED** [review]

**Review by JonathanHuot — APPROVED** [review]
LGTM

File: requests_oauthlib/compliance_fixes/ebay.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def ebay_compliance_fix(session):
+    def _compliance_fix(response):
+        token = json.loads(response.text)
+
+        # eBay responds with non-compliant token types.
+        # https://developer.ebay.com/api-docs/static/oauth-client-credentials-grant.html
+        # https://developer.ebay.com/api-docs/static/oauth-auth-code-grant-request.html
+        # Modify these to be "Bearer".
+        if token["token_type"] in ["Application Access Token", "User Access Token"]:
```
Comment (craiga) [inline requests_oauthlib/compliance_fixes/ebay.py:13]: Oops, token_type isn't always present. This should be .get…

**Review by jtroussard — APPROVED** [review]
lgtm

**Comment by jtroussard** [comment]
Note: verified unit tests locally (passes), pipeline (missing coverage report) probably needs to be reconfigured. Will create issue and look into.

**Pull request merged**